p2p/protocol/identify/id.go

```diff
@@ -143,13 +150,24 @@ func NewIDService(ctx context.Context, h host.Host, opts ...Option) *IDService {
 	return s
 }
 
+// Close shuts down the IDService
+func (ids *IDService) Close() error {
+	ids.closeSync.Do(func() {
+		ids.ctxCancel()
+		ids.refCount.Wait()
+	})
+	return nil
+}
+
 func (ids *IDService) handleEvents() {
+	ids.refCount.Add(1)
 	sub := ids.subscription
 	defer func() {
 		_ = sub.Close()
 		// drain the channel.
 		for range sub.Out() {
```

Comment:
note: we no longer need this (close will do it for us). We can simplify to:

```go
defer ids.refCount.Done()
defer sub.Close()
```

Reply:
Done.